### PR TITLE
fix(skills): surface hub import error details

### PR DESCRIPTION
**Pull request by fancyboi999**
## Summary

This PR makes skill hub import failures much easier to understand by preserving backend error detail across the full request path.

## Root Cause

The skill import flow had two issues:

1. Re-importing an existing skill raised a generic `RuntimeError`, which the router converted into `502 Bad Gateway`, even though this is a normal business conflict rather than an upstream failure.
2. The console frontend discarded backend error details and always showed a generic `Import failed` toast.

## Changes

- Return `409 Conflict` for duplicate skill imports with a clear `detail` message telling the user to retry with `overwrite=true`.
- Preserve backend `detail` values in the console request layer by extracting JSON error payloads instead of surfacing raw `Request failed: ...` blobs.
- Show the real error message in the Skills import UI instead of a hard-coded `Import failed` toast.
- Add router tests covering duplicate-import `409` and invalid-url `400` behavior.

## Validation

Passed:

- `PYTHONPATH=src python3 -m pytest tests/unit/app/routers/test_skills.py`
- `cd console && npx eslint src/api/request.ts src/pages/Agent/Skills/useSkills.ts`

Not fully runnable in this workspace:

- `cd console && npm run build:test`
  - Fails due to existing missing frontend dependencies unrelated to this change:
    - `@dnd-kit/sortable`
    - `@dnd-kit/utilities`
    - `@dnd-kit/core`

## Issue

Refs #1204

### CODE REVIEW

**Comment by fancyboi999** [comment]
Follow-up update:

- addressed the CI failure from the first revision (`# -*- coding: utf-8 -*-` + test formatting)
- hardened hub import name handling to reject unsafe bundle skill names before they are used as filesystem paths
- covered the unsafe-name case with an extra unit test
- preserved the duplicate-import `409` behavior even when `create_skill()` returns `False` after the existence check

Current status on the latest commit:
- `Pre-commit Checks`: passed
- `console format check`: passed
- `website format check`: passed
- `Tests`: still waiting on maintainer approval gate

File: src/copaw/agents/skills_hub.py [hunk 1]
```diff
@@ -1157,6 +1157,11 @@ def install_skill_from_hub(
         fallback = urlparse(bundle_url).path.strip("/").split("/")[-1]
         name = _safe_fallback_name(fallback)
 
+    if not overwrite and (get_customized_skills_dir() / name).exists():
+        raise FileExistsError(
+            f"Skill '{name}' already exists. Use overwrite=true to replace.",
+        )
+
     created = SkillService.create_skill(
```
[the remark this answers is not in the document]
Comment (fancyboi999) [reply src/copaw/agents/skills_hub.py:1181]: Handled in the follow-up commit. The duplicate-import path now re-checks the target directory after `create_skill()` returns `False`, so concurrent conflicts are translated to `FileExistsError` and still surface as `409` instead of falling through to `502`.

File: src/copaw/app/routers/skills.py [hunk 2]
```diff
@@ -136,6 +136,14 @@ async def install_from_hub(request: HubInstallRequest):
...
             overwrite=request.overwrite,
         )
+    except FileExistsError as e:
+        detail = str(e)
+        logger.warning(
+            "Skill hub install 409: bundle_url=%s detail=%s",
+            (request.bundle_url or "")[:80],
+            detail,
+        )
+        raise HTTPException(status_code=409, detail=detail) from e
```
[the remark this answers is not in the document]
Comment (fancyboi999) [reply src/copaw/app/routers/skills.py:146]: Addressed by the same name-validation change. Since unsafe bundle names are rejected before the existence check, the duplicate-import `409` detail no longer exposes attacker-controlled path fragments.

File: src/copaw/agents/skills_hub.py [hunk 3]
```diff
@@ -1157,6 +1157,11 @@ def install_skill_from_hub(
         fallback = urlparse(bundle_url).path.strip("/").split("/")[-1]
         name = _safe_fallback_name(fallback)
 
+    if not overwrite and (get_customized_skills_dir() / name).exists():
```
[the remark this answers is not in the document]
Comment (fancyboi999) [reply src/copaw/agents/skills_hub.py:1176]: Handled in the follow-up commit. Hub-provided skill names are now validated before any filesystem use, and unsafe names containing absolute-path/path-traversal patterns are rejected with `400`.

[the remark this answers is not in the document]
Comment (fancyboi999) [reply src/copaw/agents/skills_hub.py:1176]: Handled in the follow-up commit. The imported skill name is now validated before any path construction, so hub bundle names can no longer escape the customized skills directory.

File: tests/unit/app/routers/test_skills.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+) -> None:
+    def _raise_value_error(**_kwargs):
+        raise ValueError("bundle_url must be a valid http(s) URL")
+
+    monkeypatch.setattr(
+        skills_router_module,
+        "install_skill_from_hub",
+        _raise_value_error,
+    )
+
```
[the remark this answers is not in the document]
Comment (fancyboi999) [reply tests/unit/app/routers/test_skills.py]: Applied in the follow-up commit. This test now exercises the real invalid-URL path instead of monkeypatching the implementation.

File: tests/unit/app/routers/test_skills.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+
+from copaw.app.routers import skills as skills_router_module
```
[the remark this answers is not in the document]
Comment (fancyboi999) [reply tests/unit/app/routers/test_skills.py:7]: Fixed in the follow-up commit. The test file now matches the repo header convention and the initial pre-commit failure is gone on the latest revision.

**Comment by xieyxclack** [comment]
Closed as the issue has been resolved
